### ISSUE TITLE
Clock corrections fail to parse for Parkes TOAs

### ISSUE THREAD

**Issue by paulray** [nanograv/PINT#88]
Looks  like the clock correction parser fails when trying to read $TEMPO/clock/time_pks.dat
```
In [5]: toas.apply_clock_corrections()
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-33-16b1cf8a7638> in <module>()
----> 1 toas.apply_clock_corrections()

/Users/paulr/src/PINT/pint/toa.pyc in apply_clock_corrections(self)
    535             if (key['obs'] in observatories and key['obs'] != "Geocenter"):
    536                 mjds, ccorr = obsmod.get_clock_corr_vals(key['obs'])
--> 537                 tvals = numpy.array([t.mjd for t in grp['mjd']])
    538                 if numpy.any((tvals < mjds[0]) | (tvals > mjds[-1])):
    539                     # FIXME: check the user sees this! should it be an exception?

/Users/paulr/src/PINT/pint/observatories.pyc in get_clock_corr_vals(obsname, **kwargs)
     41     # of normal TEMPO-style clock correction files
     42     mjds, ccorr = numpy.loadtxt(filenm, skiprows=2,
---> 43                                 usecols=(0, 2), unpack=True)
     44     return mjds, ccorr
     45 

/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/numpy/lib/npyio.pyc in loadtxt(fname, dtype, comments, delimiter, converters, skiprows, usecols, unpack, ndmin)
    926 
    927             # Convert each value according to its column and store
--> 928             items = [conv(val) for (conv, val) in zip(converters, vals)]
    929             # Then pack it according to the dtype's nesting
    930             items = pack_items(items, packing)

/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/numpy/lib/npyio.pyc in floatconv(x)
    657         if b'0x' in x:
    658             return float.fromhex(asstr(x))
--> 659         return float(x)
    660 
    661     typ = dtype.type

ValueError: could not convert string to float: H
```

It seems to be trying to read columns (0,2) from the clock correction file, but that is not OK for the Parkes file. Here are the first few lines of that file.  It dies when it tries to convert 'H' to a float.
```
    MJD       PKS-TID  GPS-PKS  NSITE  Date and comments

  44000.0       0.0               2    ATCA
  60000.0       0.0               2    ATCA
  44000.0       0.0               H    JBmk2
  60000.0       0.0               H    JBmk2
  44000.0       0.0               E    MOST
  60000.0       0.0               E    MOST
 44000.00       0.092             7    PKS
 48811.00       2.033             7    080792
 48812.00       2.128             7    090792
 48813.00       2.296             7    100792
 48814.00       2.396             7    110792
 48815.00       2.492             7    120792
```

**Comment by luojing1211** [comment]
I see the same problem. We should update clock correction reading code.

**Comment by paulray** [comment]
Of course, this is really just a reminder that we should figure out a better way to do clock corrections for PINT than reading those ancient format files from $TEMPO/clock

**Comment by luojing1211** [comment]
By looking at this file, I would like to ask, what is NSITE mean and what is 'H' or 'E' doing there. Currently when we read in the file, we are simply use numpy.loadtext. I think that is where we are failing reading old pks clock correction. 

On the other hand, in TEMPO2, does Tempo style clock correction get abandoned. I can not find them in the TEMPO2 runtime folder.

**Comment by paulray** [comment]
A simple fix is in pull request #90 
However, I think we should pull the clock correction code out of toa.py into its own file, and make it more generic. It would be good to have an interface that would allow the user to specify whether to use TEMPO clock files, Tempo2 corrections, or some other PINT-specific method that we come up with.
In general, I think the clock correction process should be more explicitly specified, and thus less easy for users to ignore.
